src/System.Net.Security/src/System/Net/ContextFlagsAdapterPal.Unix.cs

```diff
@@ -32,9 +30,23 @@ public ContextFlagMapping(Interop.NetSecurityNative.GssFlags gssFlag, ContextFla
         };
 
 
-        internal static ContextFlagsPal GetContextFlagsPalFromInterop(Interop.NetSecurityNative.GssFlags gssFlags)
+        internal static ContextFlagsPal GetContextFlagsPalFromInterop(Interop.NetSecurityNative.GssFlags gssFlags, bool isServer)
         {
             ContextFlagsPal flags = ContextFlagsPal.Zero;
+
+            // GSS_C_INTEG_FLAG is handled separately as its value can either be AcceptIntegrity (used by server) or InitIntegrity (used by client)
+            if ((gssFlags & Interop.NetSecurityNative.GssFlags.GSS_C_INTEG_FLAG) != 0)
+            {
+                if (isServer)
+                {
+                    flags |= ContextFlagsPal.AcceptIntegrity;
+                }
+                else
+                {
+                    flags |= ContextFlagsPal.InitIntegrity;
+                }
```

Comment:
What you have is fine, but this could be condensed to:
```C#
flags |= isServer ? contextFlagsPal.AcceptIntegrity : ContextFlagsPal.InitIntegrity;
```